### DISCUSSION TITLE
Sonoff 4CH R2: 3 Water Valve + 1  leakage sensor or contact sensor

### DISCUSSION THREAD

**Discussion by Sergbmw** [RavenSystem/esp-homekit-devices#2084] [MEPLHAA Scripts]
Prompt configuration
We have sonoff 4h pro
relay 1-3 - irrigation valve opening
relay 4 or GPIO use as leakage sensor or contact sensor

And tell me the connection diagram of the contact sensor

**Comment by Sergbmw** [comment]
`{"c":{"ct":28,"io":[[[4,5,12,13],2],[[0,9,10,14],6],[[15],7,0]],"l":13},"a":[{"t":20,"w":1,"0":{"r":[[15]]},"1":{"r":[[15,1]]},"b":[[14]],"es":[{"t":20,"w":1,"0":{"r":[[12]]},"1":{"r":[[12,1]]},"b":[[0]]},{"t":20,"w":1,"0":{"r":[[5]]},"1":{"r":[[5,1]]},"b":[[9]]},{"t":20,"w":1,"0":{"r":[[4]]},"1":{"r":[[4,1]]},"b":[[10]]}]}}
`

Not view sensor

**Comment by Sergbmw** [comment]
Will someone answer me?

**Reply by PhaseAngle** [reply]
There is no access to GPIO on the Sonoff 4CH R2 the outputs are relay contacts

**Reply by Sergbmw** [reply]
Are there other connection options?

**Reply by PhaseAngle** [reply]
Either use a second device like an esp01 for the sensor or change to something else that has relays and inputs.
